trs_filer/ga4gh/trs/endpoints/register_objects.py

```diff
@@ -392,6 +391,14 @@ def process_files(self) -> None:
                         )
                         raise BadRequest
                     self.primary_descriptor_flags[_file['type']] = True
+
+                if _file['tool_file']['file_type'] == "SECONDARY_DESCRIPTOR":
+                    if not self.primary_descriptor_flags[_file['type']]:
+                        logger.error(
+                            "PRIMARY_DESCRIPTOR must be added for the same "
```

Comment:
In my opinion, we need to check the whole array of descriptor files. Because it a possible that a person inputs in a random order. For example `[SECONDARY_DESCRIPTOR, SECONDARY_DESCRIPTOR, PRIMARY_DESCRIPTOR]`, which I think is a valid case, but will still raise a `BadRequest` as per this implementation. We need to troubleshoot cases such as `[SECONDARY_DESCRIPTOR, SECONDARY_DESCRIPTOR]`, for the problem at hand.

Reply:
Correct

Reply:
I'll try to fix it

Reply:
That should be how it is implemented now? Let me know if anything needs to change.